includes/class-base-widget.php

```diff
@@ -397,9 +397,126 @@ protected function render_form_textarea( array $field ) {
 	}
 
 	/**
+	 * Render the hours container and select fields
+	 *
+	 * @param  array $field Field data.
+	 * @param  array $day   The current day in the iteration.
+	 * @param  array $hours The array of times.
+	 *
+	 * @since NEXT
+	 *
+	 * @return mixed
+	 */
+	protected function render_day_input( $field, $day, array $hours ) {
+
+		$field['name']     = str_replace( 'value', strtolower( $day ), $field['name'] );
+		$field['disabled'] = $hours['not_open'] ? true : false;
+
+		$apply_to_all_toggle = key( $field['days'] ) === $day ? '<a href="#" class="js_wpcw_apply_hours_to_all">' . sprintf( _x( 'Apply to All %s', 'Dashicon Down Arrow', 'contact-widgets' ), '<span class="dashicons dashicons-arrow-down-alt"></span>' ) . '</a>' : '';
+
+		$custom_text_checkbox = '<input name="' . $field['name'] . '[custom_text_checkbox]" id="' . $field['name'] . '[custom_text_checkbox]" class="js_wpcw_custom_text_checkbox" type="checkbox" value="1" ' . $this->checked( $hours['custom_text_checkbox'], true ) . '><label for="' . $field['name'] . '[custom_text_checkbox]" class="js_wpcw_custom_text_checkbox"><small>' . esc_html__( 'Custom Text', 'contact-widgets' ) . '</small></label>';
+		$hidden_text_field    = ( ! $hours['custom_text_checkbox'] ) ? ' style="display:none;"' : '';
+		$custom_text_field    = '<input type="text" name="' . $field['name'] . '[custom_text]" id="' . $field['name'] . '[custom_text]" class="widefat custom_text_field" value="' . $hours['custom_text'] . '"' . $hidden_text_field . '/>';
+		$closed_checkbox      = '<input name="' . $field['name'] . '[not_open]" id="' . $field['name'] . '[not_open]" class="js_wpcw_closed_checkbox" type="checkbox" value="1" ' . $this->checked( $hours['not_open'], true ) . '><label for="' . $field['name'] . '[not_open]" class="js_wpcw_closed_checkbox"><small>' . esc_html__( 'Closed', 'contact-widgets' ) . '</small></label>';
+
+		printf(
+			'<div class="day-container">%1$s<p>%2$s %3$s</p></div>',
+			'<strong>' . esc_html( ucwords( $day ) ) . '<span class="day-checkbox-toggle">' . $apply_to_all_toggle . $custom_text_checkbox . $closed_checkbox . '</strong></span>',
+			$this->render_hours_selection( $field, sanitize_title( $day ), $hours ),
+			$custom_text_field
+		);
+
+	}
+
+	/**
+	 * Render the 'Hours' select fields
+	 *
+	 * @param array $field Field data
+	 * @param array $day   The current day in the iteration.
+	 * @param array $hours The array of times.
+	 *
+	 * @since NEXT
+	 *
+	 * @return mixed
+	 */
+	protected function render_hours_selection( $field, $day, $hours ) {
+
+		ob_start();
+
+		$times = $this->get_half_hour_time_array();
+
+		$field['name'] = str_replace( 'value', strtolower( $day ), $field['name'] );
+
+		$disabled_field = $field['disabled'] ? ' disabled="disabled"' : '';
+		$hidden_field   = $hours['custom_text_checkbox'] ? ' style="display:none;"' : '';
```

Comment:
@jonathanbardo Looking back on this it may be best to utilize classes for hidden. I'm not a fan of inline styles. Thoughts?